PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/sparkserverless/serverexplore/ui/SparkServerlessProvisionDialog.java

```diff
@@ -89,73 +95,89 @@ public SparkServerlessProvisionDialog(@NotNull SparkServerlessADLAccountNode adl
         this.ctrlProvider = new SparkServerlessClusterProvisionCtrlProvider(
                 this, new IdeaSchedulers((Project)adlAccountNode.getProject()), account);
         this.adlAccountNode = adlAccountNode;
-        this.account = account;
 
         init();
         this.setTitle("Provision Spark Cluster");
         errorMessageField.setBackground(this.provisionDialogPanel.getBackground());
         errorMessageField.setBorder(BorderFactory.createEmptyBorder());
+        availableAUField.setBorder(BorderFactory.createEmptyBorder());
+        totalAUField.setBorder(BorderFactory.createEmptyBorder());
+        calculatedAUField.setBorder(BorderFactory.createEmptyBorder());
         this.setModal(true);
 
+        // setClusterNameSets to enable cluster name uniqueness check
+        this.setClusterNameSets();
+        // We can determine the ADL account since we provision on a specific ADL account Node
+        this.adlAccountField.setText(adlAccountNode.getAdlAccount().getName());
+        this.storageRootPathLabel.setText(Optional.ofNullable(account.getStorageRootPath()).orElse(""));
+
         adlAccountField.addFocusListener(new FocusAdapter() {
             @Override
             public void focusLost(FocusEvent e) {
-                ctrlProvider.updateAvailableAU();
-                ctrlProvider.updateTotalAU();
+                updateAvailableAU();
+                updateTotalAU();
             }
         });
-
-        refreshButton.addActionListener(e -> ctrlProvider.updateAvailableAU());
-
-        Arrays.asList(
-                masterCoresField, workerCoresField, masterMemoryField, workerMemoryField, workerNumberOfContainersField)
+        refreshButton.addActionListener(e -> updateAvailableAU());
+        Arrays.asList(masterCoresField, workerCoresField, masterMemoryField, workerMemoryField,
+                workerNumberOfContainersField)
```

Comment:
Missing two fields?

Reply:
updateCalculdaedAU is only triggered when these five fields changes. Cluster name and spark events have no relationship with calculatedAU field.

---
In reply to: [199446945](https://github.com/Microsoft/azure-tools-for-java/pull/1747#discussion_r199446945) [](ancestors = 199446945)